packages/app/src/components/MenuBar/MenuBar.js

```diff
@@ -77,8 +78,10 @@ const mapStateToProps = state => ({
 const mapDispatchToProps = dispatch => ({
     onRenameVisualization: details =>
         dispatch(fromActions.tDoRenameVisualization(details)),
-    onSaveVisualization: (details, copy) =>
-        dispatch(fromActions.tDoSaveVisualization(details, copy)),
+    onSaveVisualization: (details, copy) => {
+        dispatch(fromLoader.acSetPluginLoading(true))
```

Comment:
What about moving this to the `tDoSaveVisualization` function? That way the flag is more centrally controlled as both the turning on and the turning off of the flag are in the same place.

Note: 
There seems to be an unused redux prop called `isLoading` (acSetIsLoading). Maybe that is the more appropriate name for this flag, since it is used not only for the plugin loading state, but also the api request? Also, I see that tDoLoadVisualization also sets the acSetPluginLoading flag, but does so after the api request (https://github.com/dhis2/data-visualizer-app/blob/master/packages/app/src/actions/index.js#L56), unlike tDoSaveVisualization, which sets the flag before the request. Maybe these should behave the same? And maybe tDoSaveVisualization should also set the flag to false in the catch. What do you think? I know this grows the scope a tiny bit, but it all kind of hangs together.

Reply:
Do you mean `sGetIsLoading` and `acSetLoading`. I initially thought that this was used to control a loading spinner for the full page, but it turns out that neither of these are actually in use (not that I can find in the code in DV anyway). If that's the case then `acSetPluginLoading` could become `acSetLoading` instead.

Reply:
I agree with moving the loading call to `tDoSaveVisualization`, updated

Reply:
Yes `sGetIsLoading` was the one I was talking about.